### PR TITLE
[WIP] [wasm] AOT work.

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/builtin-types.cs [hunk 1]
```diff
@@ -382,7 +382,7 @@ public virtual nint GetAVirtual ()
 		}
 	}
 
-	public int test_0_nint_fieldload ()
+	static int test_0_nint_fieldload ()
```
Comment (lewurm) [inline mono/mini/builtin-types.cs]: ugh, sorry about that

**Comment by vargaz** [comment]
@monojenkins build Linux WebAssembly

**Comment by vargaz** [comment]
@monojenkins build Linux WebAssembly

**Comment by vargaz** [comment]
@monojenkins build Linux WebAssembly

File: sdks/wasm/packager.cs [hunk 2]
```diff
@@ -370,7 +370,7 @@ enum AssemblyKind {
 		ninja.WriteLine ("build $appdir: mkdir");
 		ninja.WriteLine ("build $appdir/$deploy_prefix: mkdir");
 		ninja.WriteLine ("build $appdir/runtime.js: cpifdiff $builddir/runtime.js");
-		ninja.WriteLine ("build $appdir/config.js: cpifdiff $builddir/config.js");
+		ninja.WriteLine ("build $appdir/mono-config.js: cpifdiff $builddir/mono-config.js");
```
Comment (kumpera) [inline sdks/wasm/packager.cs]: why?

**Comment by vargaz** [comment]
@monojenkins build Linux WebAssembly

**Comment by vargaz** [comment]
@monojenkins build Linux WebAssembly

**Comment by vargaz** [comment]
@monojenkins build failed

**Comment by vargaz** [comment]
@monojenkins build Linux x64

File: sdks/wasm/driver.c [hunk 3]
```diff
@@ -249,10 +120,10 @@ mono_wasm_add_assembly (const char *name, const unsigned char *data, unsigned in
 {
 	int len = strlen (name);
 	if (!strcasecmp (".pdb", &name [len - 4])) {
-		name = m_strdup (name);
+		char *new_name = m_strdup (name);
```
Comment (jaykrell) [inline sdks/wasm/driver.c]: Overlaps https://github.com/mono/mono/pull/11088 but ok.

**Comment by jaykrell** [comment]
wasm/C++ regression regarding CFLAGS or CXXFLAGS? Please fix but don't block the PR?

```
driver.c:7:10: fatal error: 'mono/metadata/assembly.h' file not found
#include <mono/metadata/assembly.h>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
```

**Pull request merged**